src/osemgrep/networking/Rule_fetching.ml

```diff
@@ -112,13 +115,21 @@ let fetch_content_from_url_async ?(token_opt = None) (url : Uri.t) :
     in
     let%lwt res = Http_helpers.get_async ?headers url in
     match res with
-    | Ok body -> Lwt.return body
+    | Ok body ->
+        (* Mark our spinner as completed *)
+        spinner_stopper := true;
+        (* Flush any pending output to prevent surprises *)
+        let%lwt () = Lwt_io.flush Lwt_io.stdout in
+        if !ANSITerminal.isatty Unix.stdout then
+          (* Clean up the loading indicator *)
+          Console_Spinner.erase_spinner ();
+        (* Return the actual http response *)
+        Lwt.return body
```

Comment:
we shouldn't be doing anything cli related in this library, it will break the language server, and also doesn't make sense organizationally. maybe have some sort of optional hook parameter for this function that does this stuff and is cli agnostic

Reply:
true

Reply:
maybe add an argument to the function, or if the call stack is too deep, add a "hook".